### ISSUE TITLE
Cannot create payment channel 

### ISSUE THREAD

**Issue by marcosmartinez7** [raiden-network/raiden#3445]
## Intro 

Discovered this issue into master branch when block number is arround 100k into GETH private network.

The env is basically one geth minner running alone and 2 nodes of raiden using it

Basically there is a query of etg_getLogs from 0 to latest (in this case from 0 to 100k) that takes more than 10 seconds so the default timeout of web3.py fails (10 seconds)

Making the web3 timeout bigger will make no sense because the response time is related with the number of blocks of the chain.

## Issue checklist

To help us more easily debug your issue please check the boxes for which requirements for safe usage or Raiden were followed:

- [x] The Ethereum node was always up to date with the chain tip and never stopped running.
- [x ] No Ethereum transaction was sent for the account that Raiden manages by another app.
- [ x] The Ethereum account used by Raiden always had sufficient ETH.
- [ ] The Raiden DB that exists by default at ~/.raiden was never altered or deleted.
- [ ] The Raiden node was never restarted. If yes please provide more info.
- [x ] The ethereum node was not switched while Raiden was running.
- [ x] The Raiden REST API is protected from public access.

-->

## Problem Definition

Basically, i was experimenting some timeouts when raiden invokes `eth.getLogs`, and after some debugging i realize something that i understand is a problem. 

- Private geth network
- Node A creates a new payment channel with node B 
- Node B raiden code creates the PaymentChannel object and check for the logs of the payment channel
- The private get network was running for some time, so the block number is 100.000
- The raiden code seems to check from block 0 to 'latest'
- That rpc call is to slow, it takes more than the 10 seconds timeout of web3

This is the trace

```
urllib3.exceptions.ReadTimeoutError: HTTPConnectionPool(host='198.211.99.46', port=4444): Read timed out. (read timeout=10)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "src/gevent/greenlet.py", line 716, in gevent._greenlet.Greenlet.run
  File "/root/raiden/raiden/raiden/tasks.py", line 117, in _run
    self.loop_until_stop()
  File "/root/raiden/raiden/raiden/tasks.py", line 157, in loop_until_stop
    self._maybe_run_callbacks(latest_block)
  File "/root/raiden/raiden/raiden/tasks.py", line 222, in _maybe_run_callbacks
    result = callback(latest_block)
  File "/root/raiden/raiden/raiden/raiden_service.py", line 558, in _callback_new_block
    on_blockchain_event(self, event)
  File "/root/raiden/raiden/raiden/blockchain_events_handler.py", line 312, in on_blockchain_event
    handle_channel_new(raiden, event)
  File "/root/raiden/raiden/raiden/blockchain_events_handler.py", line 79, in handle_channel_new
    channel_identifier,
  File "/root/raiden/raiden/raiden/network/blockchain_service.py", line 209, in payment_channel
    contract_manager=self.contract_manager,
  File "/root/raiden/raiden/raiden/network/proxies/payment_channel.py", line 44, in __init__
    events = token_network.proxy.contract.web3.eth.getLogs(filter_args)
```

And this is the part of the code that i think is problematic: 

Class payment_channel.py

```python
class PaymentChannel:
    def __init__(
            self,
            token_network: TokenNetwork,
            channel_identifier: ChannelID,
            contract_manager: ContractManager,
    ):

        self.contract_manager = contract_manager
        if channel_identifier < 0 or channel_identifier > UINT256_MAX:
            raise ValueError('channel_identifier {} is not a uint256'.format(channel_identifier))

        filter_args = get_filter_args_for_specific_event_from_channel(
            token_network_address=token_network.address,
            channel_identifier=channel_identifier,
            event_name=ChannelEvent.OPENED,
            contract_manager=self.contract_manager,
        )

        events = token_network.proxy.contract.web3.eth.getLogs(filter_args)
```

That function takes GENESIS_BLOCK_NUMBER (0) as default, so, it makes a query from 0 to 100k

```python
def get_filter_args_for_specific_event_from_channel(
        token_network_address: TokenNetworkAddress,
        channel_identifier: ChannelID,
        event_name: str,
        contract_manager: ContractManager,
        from_block: BlockSpecification = GENESIS_BLOCK_NUMBER,
        to_block: BlockSpecification = 'latest',
):

```

**Comment by marcosmartinez7** [comment]
Nevermind, seems more a onchain issue, the ethlogs is too slow, not a Raiden issue. 

Could you guys validate that raiden does make eth_getLogs requests from 0 to 'latest' with regularity and it is a normal behaviour?

**Comment by LefterisJP** [comment]
the `0` to `latest` will only happen when a token network is initialized. But let me have a look as something looks funny there. We have more data so that we don't need to start from `0`, we can at times start from the token network's deployment block.

**Comment by LefterisJP** [comment]
@marcosmartinez7 Okay so I refreshed my memory.

Yes indeed we do regurarly query events from `0` to `latest`. For example in the settle timeout query:

https://github.com/raiden-network/raiden/blob/66f15f21a17a7b5080a9f48e4010cf158d7ad089/raiden/network/proxies/payment_channel.py#L80-L100

This can be improved if we keep at least the token network registry's deployment block. We got an issue [here](https://github.com/raiden-network/raiden/issues/1129#issuecomment-392733610) for it. But since this has not yet been identified as a bottleneck that issue has not been worked on.
